### PR TITLE
Adding DaaS V2 API

### PR DESCRIPTION
**Pull request by puneetg1983**
For the last few weeks, I have been working on writing V2 for DAAS which :-
-	Should make DAAS session schema exactly like Linux so we can have a common API path for both Linux and Windows. 
-	Wanted to simplify the DaaS code as it super legacy.
-	Remove XML dependency and make all the session and configuration files as JSON as XML has given too many problems.

For some time, we will have both V1 and V2 for DAAS in parallel and once the UI gets fully switched to use V2, I will simply remove V1 code.

Along with this PR, we are also adding binaries for DiagnosticsAnalysis.exe (a tool that Visual Studio diagnostics team wants us to run along with memory dump analysis). Currently we are invoking the tool and just logging the telemetry in Kusto. The output of this tool against a memory dump is something like this

`{"Dump":"dw0sdwk0000LU_w3wp_18176.dmp","Analysis":{"interpretedResults":[{"shortDescription":"Thread CPU usage summary","severity":"info","errorCode":"AA0020"},{"shortDescription":"Exception objects were found in the heap","severity":"info","errorCode":"AA0013"}]}}
`

You may notice a lot of files changed in the PR but that’s basically I added a .editorconfig file to fix the file headers everywhere and that has added an extra space.

### CODE REVIEW

**Comment by hforeste** [comment]
@puneetg1983 417 files 🤣

File: ClrProfilingAnalyzer/Parser/AspNetCoreRequestParser.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
-//-----------------------------------------------------------------------
+// -----------------------------------------------------------------------
 // <copyright file="AspNetCoreRequestParser.cs" company="Microsoft Corporation">
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 // </copyright>
-//-----------------------------------------------------------------------
+// -----------------------------------------------------------------------
```
Comment (ShekharGupta1988) [inline ClrProfilingAnalyzer/Parser/AspNetCoreRequestParser.cs:6]: nice ;) I love the change

File: DaaS/V2/Configuration/Diagnoser.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public string ProcessCleanupOnCancel { get; set; }
+        public bool RequiresStorageAccount { get; set; }
+        public CollectorConfiguration Collector { get; set; }
+        public AnalyzerConfiguration Analyzer { get; set; }
+
+        public List<string> GetWarnings()
+        {
+            var warnings = new List<string>();
+            var collector = new Collector(this);
+            if (!string.IsNullOrEmpty(collector.Warning))
```
Comment (ShekharGupta1988) [inline DaaS/V2/Configuration/Diagnoser.cs:25]: nit : whitespace :)

Comment (hforeste) [reply DaaS/V2/Configuration/Diagnoser.cs:25]: @ShekharGupta1988  Do you have a regex browser extension that looks for these? 🤣

File: DaaS/V2/Diagnostics/PredefinedValidators.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            {
+                //
+                // Try to get the path of rt.jar by running KuduHandles on java.exe
+                //
+
+                var rtJarHandle = GetJarFileHandle(javaProcess.Id);
+                if (!string.IsNullOrWhiteSpace(rtJarHandle) && rtJarHandle.Length > 0)
+                {
+                    var parentPath = rtJarHandle.Replace(@"\jre\lib\rt.jar", "").Replace("c:", "d:");
+                    var javaToolPath = Path.Combine(parentPath, "bin", toolName);
```
Comment (ShekharGupta1988) [inline DaaS/V2/Diagnostics/PredefinedValidators.cs:119]: I see a pattern here in the code where you are checking that if a file exists (for the tool), you return true else you have an error message and return false. Do you feel repetitive code can be reduced by having a method something like :

`bool CheckToolExists(string path, out errorMsg);`

You can then simply use it like this:

`return CheckToolExists(javaToolPath, out AdditionalError);`

File: DaaS/V2/Configuration/Diagnoser.cs (hunk 2 above)
Comment (ShekharGupta1988) [reply DaaS/V2/Configuration/Diagnoser.cs:25]: hehe..

Comment (puneetg1983) [reply DaaS/V2/Configuration/Diagnoser.cs]: done...  😄

File: DaaS/V2/Diagnostics/PredefinedValidators.cs (hunk 3 above)
Comment (puneetg1983) [reply DaaS/V2/Diagnostics/PredefinedValidators.cs]: Nice suggestion. Implemented!

File: DaaS/V2/Utility.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        /// <summary>
+        /// The functionality in Antares to get the HostName is a bit flaky. This method is used mostly to construct
+        /// the paths of logs and reports in the session and if we fail to identify the hostname properly, we will
+        /// return empty and let the client (UI or CLI) append the SCM hostname to the URL
+        /// </summary>
+        /// <returns>The default hostname of the site or an empty string</returns>
+        public static string GetScmHostName()
+        {
+            var scmHostName = Settings.Instance.DefaultScmHostName;
+            if (!string.IsNullOrEmpty(scmHostName)
```
Comment (ShekharGupta1988) [inline DaaS/V2/Utility.cs:23]: I dont know why these catches my eye :D

File: DiagnosticAnalysisLauncher/Models/DiagnosticAnalysis.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+{
+    public class DiagnosticAnalysis
+    {
+        public Interpretedresult[] interpretedResults { get; set; }
+
+        [JsonIgnore]
+        public Result[] results { get; set; }
+    }
+
+    public class Interpretedresult
```
Comment (ShekharGupta1988) [inline DiagnosticAnalysisLauncher/Models/DiagnosticAnalysis.cs:20]: small nitpick in naming convention. Should be InterpretedResult

**Review by ShekharGupta1988 — APPROVED** [review]

File: DiagnosticAnalysisLauncher/Models/DiagnosticAnalysis.cs (hunk 5 above)
Comment (puneetg1983) [reply DiagnosticAnalysisLauncher/Models/DiagnosticAnalysis.cs:20]: Yes unfortunately. This is an auto-generated class based on the json model generated by Visual Studio Diagnostic tool itself and we cannot change the casing here.

File: DaaS/V2/Utility.cs (hunk 4 above)
Comment (puneetg1983) [reply DaaS/V2/Utility.cs]: done.

**Pull request merged**